src/shlink.ts

```diff
@@ -0,0 +1,56 @@
+import { ErrorCode } from "./error";
+import Log from "./logger";
+import { IOptions } from "./options";
+import * as shlPayload from "./shlPayload";
+
+export async function validate(shlinkText: string, options: IOptions): Promise<Log> {
+    const log = new Log("SMART Health Link");
+
+    const regEx = /^(\s*)?(.+?#)?(.+?)\/?([0-9a-zA-Z-_+/]+)(\s*)?$/;
+
+    const shlinkHeader = "shlink:";
+
+    const [match, leadingWhiteSpace, viewer, header, body, trailingWhiteSpace] = [...(regEx.exec(shlinkText) ?? [])];
+
+    if (viewer) log.info(`Viewer: ${viewer}`);
+
+    log.debug(`${JSON.stringify({ match: !!match, leadingWhiteSpace : !!leadingWhiteSpace, viewer: viewer ?? 'none', header, body, trailingWhiteSpace : !!trailingWhiteSpace }, null, 2)}`);
+
+    if (viewer && !/https:\/\/.+?#/.test(viewer)) {
+        log.error(`Unexpected viewer format`, ErrorCode.INVALID_SHLINK);
+    }
+
+    if (!match) {
+        const expectedBody = `^(https://<viewer>#)?${shlinkHeader}/[0-9a-zA-Z-_]+$`;
+        return log.fatal(`Invalid shlink : expected ${expectedBody}`, ErrorCode.INVALID_SHLINK);
+    }
+
+    if (leadingWhiteSpace || trailingWhiteSpace) {
```

Comment:
It's slightly surprising to paste a shlink into a text file (so the file has the link on a line by itself) and get a trailing whitespace warning. I can avoid this with something like `printf 'https://...' > shl.txt` but it might be better to accept a file with a single line (and a line break at the end).

Could as a quality of life thing: might be worth providing some `--path` way to accept stdin, allowing for `cat 'https://' | node . --type shlink`.

Reply:
This was how we treated SHC as we expected it in a file.  I agree we should be more forgiving with SHL since you'd just be throwing it into a file as a means to import it into the tool.

I'll look at providing some way to pipe the link in.